### ISSUE TITLE
valid-lang rule should test aria-hidden text

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3633]
### Product

axe-core

### Product Version

4.4.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

For the valid-lang rule, axe-core reports the following as passed:

```html
<article lang="english">
  <p aria-hidden="true">
    They wandered into a strange Tiki bar on the edge of the small beach town.
  </p>
</article>
```

This is a failed example of ACT, and so axe-core should fail it too. Text in aria-hidden sections aren't entirely hidden to assistive technologies. Plus there are non-AT benefits to ensuring correct language is used.

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base,
test script:
```
<article lang="english">
  <p aria-hidden="true">
    They wandered into a strange Tiki bar on the edge of the small beach town.
  </p>
</article>
```
and `valid-lang` rule for the specified script is not passing now, it is listing under failure.


![Image](https://user-images.githubusercontent.com/47409579/190316704-db75ace8-c910-4642-9234-6666a613ca76.png)
